sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/AzureJdbcPropertiesUtils.java

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.autoconfigure.jdbc;
+
+import com.azure.spring.cloud.core.implementation.properties.PropertyMapper;
+import com.azure.spring.cloud.core.properties.AzureProperties;
+import com.azure.spring.cloud.core.provider.AzureProfileOptionsProvider;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+/**
+ * Store the constants for customized Azure properties with Kafka.
+ */
+public final class AzureJdbcPropertiesUtils {
+
+    private static final PropertyMapper PROPERTY_MAPPER = new PropertyMapper();
+    static final String CREDENTIAL_PREFIX = "azure.credential.";
+    static final String PROFILE_PREFIX = "azure.profile.";
+    static final String ENVIRONMENT_PREFIX = PROFILE_PREFIX + "environment.";
+
+    public static void convertAzurePropertiesToConfigMap(AzureJdbcProperties source,
+                                                         Map<String, String> target) {
+        for (Mapping m : Mapping.values()) {
+            PROPERTY_MAPPER.from(m.getter.apply(source)).to(p -> target.putIfAbsent(m.propertyKey, p));
+        }
+    }
+
+    public static void convertConfigMapToAzureProperties(Map<String, ?> source,
+                                                         AzureJdbcProperties target) {
+        for (Mapping m : Mapping.values()) {
+            PROPERTY_MAPPER.from(source.get(m.propertyKey)).to(p -> m.setter.accept(target, (String) p));
+        }
+    }
+
+    enum Mapping {
+
+        clientCertificatePassword(CREDENTIAL_PREFIX + "client-certificate-password",
```

Comment:
what's this?
is it an enum? if so, shouldn't this be all UPPERCASE? or this some kind of hack?

Reply:
It's some kind of hack, I will have a try to refacter it